src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodOrUserDefinedOperatorSymbol.cs

```diff
@@ -238,11 +238,11 @@ internal override void AfterAddingTypeMembersChecks(ConversionsBase conversions,
             // type errors but for parameter errors, we'll use the parameter location.
             CheckConstraintsForExplicitInterfaceType(conversions, diagnostics);
 
-            this.ReturnType.CheckAllConstraints(compilation, conversions, this.Locations[0], diagnostics);
+            this.ReturnType.CheckAllConstraints(compilation, conversions, this.GetFirstLocation(), diagnostics);
```

Comment:
note: virtually all places in the compiler that do Locations[0] (and there are a lot) would benefit from this.  However, i'm trying to keep the scope of changes small, and driven by what i'm seeing in traces.